Infrastructure: remove vcpkg, boost, in favour of OS-specific package mangers

vadi2 opened the pull request:
<!-- Keep the title short & concise so anyone non-technical can understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
This change removes vcpkg and boost from the build system and switches to using platform-native package managers instead. 
#### Motivation for adding to Mudlet
The idea was to use one C++ package manager on all 3 platforms, but it's being too troublesome in practice, unfortunately. This change will simplify maintenance. 
#### Other info (issues closed, discussion etc)

claude committed:
Remove vcpkg dependency manager in favor of platform-specific package managers

This change removes vcpkg from the build system and switches to using
platform-native package managers instead:

- Linux: apt-get for system packages
- macOS: Homebrew for system packages
- Windows: Already using MSYS2 (no change needed)
- All platforms: GitHub Actions (leafo/gh-actions-lua) for Lua 5.1 in CI

Changes:
- Remove vcpkg submodule and dependency files
- Update build-mudlet.yml workflow:
  - Remove vcpkg installation and caching steps
  - Add system package installations for yajl, pcre, pugixml, sqlite, hunspell
  - Add Lua 5.1 via GitHub Actions for macOS (matching existing Linux setup)
  - Simplify lua-yajl installation (no more vcpkg path workarounds)
  - Remove vcpkg-specific luarocks configuration
- Update CMakeLists.txt: Remove vcpkg toolchain file setup
- Update README.md: Remove vcpkg from dependencies list

Benefits:
- Faster CI builds (no vcpkg source compilation)
- Simpler maintenance (no custom Lua overlay port)
- Better caching via native package managers
- Consistent with Windows builds (already vcpkg-free)
- Smaller repository (removed ~100MB+ submodule)

vadi2 commented on .github/workflows/build-mudlet.yml:
```diff
@@ -65,17 +65,13 @@ jobs:
-        # fetches everything...
-        submodules: true
+        # CMake and qmake will pull in required submodules automatically
+        submodules: recursive
```
```suggestion
        # CMake and qmake will pull in required submodules automatically
        submodules: true
```

SlySven replied:
Um, why do we do this? Both the CMake and qmake meta-project build files were written to (and should still be capable of) pulling in the required sub-modules themselves - and skipping any that are not used for the particular build (e.g. the Windows build does not need to pull in the `Sparkle` one for MacOS)?

vadi2 committed:
Apply suggestion from @vadi2

vadi2 committed:
Merge branch 'development' into claude/vcpkg-build-investigation-011CUz4EnT6cf9svh7adzkY9

claude committed:
Fix: Add system Lua package for CMake build

The leafo/gh-actions-lua is for luarocks compatibility, but CMake
needs system Lua headers and libraries to build Mudlet.

- Linux: Install liblua5.1-0-dev for CMake
- macOS: Set LUA_DIR to help CMake find gh-actions-lua installation

Both platforms still use gh-actions-lua for luarocks operations.

claude committed:
Merge branch 'claude/vcpkg-build-investigation-011CUz4EnT6cf9svh7adzkY9' of http://127.0.0.1:35441/git/vadi2/Mudlet into claude/vcpkg-build-investigation-011CUz4EnT6cf9svh7adzkY9

claude committed:
Fix: Boost detection with CMake 3.30+

CMake 3.30 introduced new policies that affect Boost detection:
- CMP0144: Changes how PACKAGENAME_ROOT variables work
- CMP0167: FindBoost module removed in favor of BoostConfig

Changes:
- Pass BOOST_ROOT and Boost_INCLUDE_DIR as CMake arguments
- Set CMP0144 policy to NEW to allow BOOST_ROOT variable usage
- Ensures header-only Boost installation is found correctly

claude committed:
Debug: Add logging to Boost extraction to diagnose layout issue

Added debug output to see:
- What gets extracted from the .7z archive
- What gets copied to the root
- What remains after cleanup

This will help identify if the boost-1.83.0.7z has a different
structure than expected.

claude committed:
Debug: Enhance Boost extraction logging

More detailed debugging to check:
- Contents of the extracted boost-1.83.0/ directory
- Whether boost-1.83.0/boost/ subdirectory exists
- Whether boost/ exists after copying

Also changed wildcard boost-*/* to explicit boost-1.83.0/* for clarity.

claude committed:
Fix: Generate Boost headers from GitHub source release

The GitHub .7z release is a source repository snapshot, not a
prepared distribution. It lacks the boost/ headers directory.

Solution: Run Boost's header generation:
- bootstrap.sh --with-libraries= (no compiled libs needed)
- b2 headers (generates boost/ directory from libs/)

The old SourceForge .tar.bz2 had headers pre-generated, but
GitHub releases require this preparation step.

claude committed:
Use system package managers for Boost instead of manual download

- Linux: Install libboost-dev via apt
- macOS: Install boost via Homebrew
- Remove manual Boost download, extraction, and header generation
- Remove BOOST_ROOT CMake arguments (system packages auto-detected)

This aligns with the vcpkg removal goal: use platform package managers
for all dependencies except Qt (which requires specific version).

claude committed:
Fix: Help CMake find gh-actions-lua installation on macOS

Pass explicit LUA_INCLUDE_DIR and LUA_LIBRARY paths to CMake on macOS
pointing to the gh-actions-lua installation at .lua/ directory.

This helps FindLua51.cmake locate the Lua headers and library that
the GitHub Action installed.

vadi2 committed:
Fix: Add PKG_CONFIG_PATH for keg-only PCRE on macOS

Homebrew's PCRE is keg-only (not symlinked to standard paths).
Set PKG_CONFIG_PATH to include /usr/local/opt/pcre/lib/pkgconfig
so FindPCRE.cmake can find the include directory via pkg-config.

vadi2 committed:
Fix: Keep PCRE installed through build, uninstall before packaging

The previous workflow uninstalled PCRE immediately after build info setup,
but CMake needs it for the actual build step.

Changes:
- Keep PCRE installed after 'Set build info' step
- Add new step to uninstall PCRE right before packaging
- Only uninstall on macOS deploy builds (where packaging happens)

This avoids the PCRE_INCLUDE_DIR missing error while still preventing
the packaging issues referenced in #6668.

vadi2 committed:
Use fix-macos-compile branch of installers repo

vadi2 committed:
Keep PCRE installed for packaging to bundle it

vadi2 committed:
Fix lua-yajl install on ARM64 macOS

vadi2 commented on .github/workflows/build-mudlet.yml:
```diff
@@ -62,19 +62,11 @@ jobs:
-        # contain conditional code to pull in the required submodules themselves
-        # except vcpkg - if that can be handled we could avoid this setting that
-        # fetches everything...
+        # CMake and qmake will pull in required submodules automatically
```
```suggestion
```

vadi2 commented on .github/workflows/build-mudlet.yml:
```diff
@@ -283,8 +206,8 @@ jobs:
-        # remove it as it breaks packaging down the line: https://github.com/Mudlet/Mudlet/issues/6668
-        brew uninstall pcre
+        # Note: Don't uninstall PCRE yet - CMake needs it for building
+        # It will be uninstalled before packaging to avoid issues (#6668)
```
```suggestion
```

vadi2 committed:
Fix YAJL_DIR detection for both x86_64 and ARM64 macOS

vadi2 committed:
Fix luarocks installs for libzip, sqlite, and pcre on macOS

vadi2 committed:
Update .github/workflows/build-mudlet.yml

vadi2 committed:
Update .github/workflows/build-mudlet.yml

vadi2 commented on CI/travis.osx.after_success.sh:
```diff
@@ -49,7 +49,7 @@ if [ "${DEPLOY}" = "deploy" ]; then
   YESTERDAY_DATE=$(date -v-1d '+%F' | tr -d '-')
 
-  git clone https://github.com/Mudlet/installers.git "${BUILD_DIR}/../installers"
+  git clone --branch fix-macos-compile https://github.com/Mudlet/installers.git "${BUILD_DIR}/../installers"
```
```suggestion
  git clone https://github.com/Mudlet/installers.git "${BUILD_DIR}/../installers"
```

vadi2 committed:
Apply suggestion from @vadi2

SlySven commented:
Looking good now! :+1:

SlySven reviewed (commented):
LGTM oh, you've already :shipit:
